services/frontend/src/scenes/Profile/ProfileLayout.js

```diff
@@ -122,18 +124,85 @@ const PersonalDetails = ({profile}) => {
     );
 };
 
-const Skills = ({sectionName, profile}) => (
-    <>
-        <h2>{sectionName}</h2>
-        <Paper className="profile-page-card profile-card-skills">
-            <SkillBadges
-                displayCount={profile.skills.length}
-                skills={profile.skills}
-            />
-        </Paper>
-    </>
-);
+const Skills = ({sectionName, profile, skills}) => {
+    const [editSkillsDialogOpen, setEditSkillsDialogOpen] = React.useState(false);
+    const [profileUpdated, updateProfile] = useState(profile);
+
+    const openDialog = () => {
+        setEditSkillsDialogOpen(true);
+    };
+
+    const closeDialog = () => {
+        setEditSkillsDialogOpen(false);
+    };
+
+    const handleSubmit = (updatedSkills) => {
+        removeSkills(updatedSkills);
+        addSkills(updatedSkills);
+        closeDialog();
+    };
+
+    const handleCancel = () => {
+        closeDialog();
+    };
+
+    const addSkills = (updatedSkills) => {
+        const tempProfile = profileUpdated;
+        updatedSkills.map((skillName) => {
+            const skillNameLC = skillName.toLowerCase();
+            if (!profileUpdated.skills.map((skill) => skill.name.toLowerCase()).includes(skillNameLC)) {
+                if (Object.values(skills).map((skill) => skill.name.toLowerCase()).includes(skillNameLC)) {
+                    /* Adds from Existing Skills List*/
+                    Object.values(skills).map((skill) => {
+                        if (skill.name.toLowerCase() === skillNameLC) {
+                            tempProfile.skills.push(skills[skill.id]);
+                        }
+                    });
+
+                } else {
+                    /*Adds new Skill Object */
+                    const newSkill = new Skill({name: skillName});
+                    tempProfile.skills.push(newSkill);
+                }
+            }
+        });
+        updateProfile(tempProfile);
+    };
+
+    const removeSkills = (updatedSkills) => {
+        const tempProfile = profileUpdated;
+        tempProfile.skills.map((skill, index) => {
```

Comment:
This is throwing warnings because you aren't returning anything from the map.

Maybe use an indexed for loop instead https://flaviocopes.com/how-to-get-index-in-for-of-loop/.